analyzer-modules/org.wso2.carbon.apimgt.throttling.siddhi.extension/src/main/java/org/wso2/carbon/apimgt/throttling/siddhi/extension/ThrottleStreamProcessor.java

```
@@ -0,0 +1,266 @@
+                                + attributeExpressionExecutors[2].getReturnType());
+            }
+        } else {
+            throw new ExecutionPlanValidationException("Throttle batch window should only have one/two parameter "
+                    + "(<int|long|time> windowTime (and <int|long> startTime), but found "
+                    + attributeExpressionExecutors.length + " input attributes");
+        }
+
+        List<Attribute> attributeList = new ArrayList<Attribute>();
+        attributeList.add(new Attribute("expiryTimeStamp", Attribute.Type.LONG));
+        return attributeList;
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+            StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+
+        synchronized (this) {
+            if (expireEventTime == -1) {
+                long currentTime = executionPlanContext.getTimestampGenerator().currentTime();
+                if (startTime != -1) {
+                    expireEventTime = addTimeShift(currentTime);
+                } else {
+                    expireEventTime = executionPlanContext.getTimestampGenerator().currentTime() + timeInMilliSeconds;
+                }
+                if (scheduler != null) {
+                    scheduler.notifyAt(expireEventTime);
+                } else {
+                    log.error("scheduler is not initiated");
+                }
+            }
+            long currentTime = executionPlanContext.getTimestampGenerator().currentTime();
+            boolean sendEvents;
+            if (currentTime >= expireEventTime) {
+                expireEventTime += timeInMilliSeconds;
+                if (scheduler != null) {
+                    scheduler.notifyAt(expireEventTime);
+                } else {
+                    log.error("scheduler is not initiated");
```

should we throw a exception?